### PR TITLE
added different learning rates for UNet and Text encoder #5271

### PR DESCRIPTION
**Pull request by shyammarjit**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Bit too specific for me tbh - @sayakpaul wdyt?

**Comment by sayakpaul** [comment]
Hmm this does actually help significantly as also found out by @linoytsaban @apolinario. They also plan on dropping a PR for this. Let's maybe wait for that one. 

But thanks very much for your efforts, @shyammarjit!

**Comment by bghira** [comment]
propagating the unet gradient across the text encoder just seems like an odd choice altogether. is there a chance of tuning the text encoder with a specific loss function that targets its abilities directly, rather than indirectly updating the weights based on how the u-net responds? it feels like the risk of overfitting is far greater when we update the text encoder weights in this way. the OpenCLIP method of fine-tuning is totally different than this.

**Comment by sayakpaul** [comment]
Thanks for your contributions. Our updated script should already have this support. Hence, I am closing this PR. But, again, I appreciate the contribution and effort put into it.
